doc/neps/nep-0031-uarray.rst

```diff
@@ -94,14 +81,60 @@ vendored into a new namespace within NumPy to give users and downstream
 dependencies access to these overrides.  This vendoring mechanism is similar
 to what SciPy decided to do for making ``scipy.fft`` overridable (see [10]_).
 
+The motivation behind ``uarray`` is manyfold: First, there have been several
+attempts to allow dispatch of parts of the NumPy API, including (most
+prominently), the ``__array_ufunc__`` protocol in NEP-13 [4]_, and the
+``__array_function__`` protocol in NEP-18 [2]_, but this has shown the need
+for further protocols to be developed, including a protocol for coercion (see
+[5]_, [9]_). The reasons these overrides are needed have been extensively
+discussed in the references, and this NEP will not attempt to go into the
+details of why these are needed; but in short: It is necessary for library
+authors to be able to coerce arbitrary objects into arrays of their own types,
+such as CuPy needing to coerce to a CuPy array, for example, instead of
+a NumPy array. In simpler words, one needs things like ``np.asarray(...)`` or
+an alternative to "just work" and return duck-arrays.
 
-Detailed description
---------------------
+Usage and Impact
+----------------
 
-Using overrides
-~~~~~~~~~~~~~~~
+This NEP allows for global and context-local overrides, as well as
+automatic overrides a-la ``__array_function__``.
 
-Here are a few examples of how an end-user would use overrides.
+Here are some use-cases this NEP would enable, besides the 
+first one stated in the motivation section:
+
+The first is allowing alternate dtypes to return their
+respective arrays.
+
+.. code:: python
+
+    # Returns an XND array
+    x = unp.ones((5, 5), dtype=xnd_dtype) # Or torch dtype
+
+The second is allowing overrides for parts of the API.
+This is to allow alternate and/or optimised implementations
+for ``np.linalg``, BLAS, and ``np.random``.
+
+.. code:: python
+
+    import numpy as np
+
+    # Makes pyfftw the default for FFT
+    import pyfftw # Or mkl_fft
+
+    # Uses pyfftw without monkeypatching
+    np.fft.fft(numpy_array)    
```

Comment:
This example is new, right? I'm not convinced that `pyfftw` should be overriding the global backend, especially since it's not always faster than `pocketfft`.

Reply:
Neither am I. But it's either a change of the backend on import, or monkeypatching. I'm not a big fan of either...

Reply:
Why can't the user just call `set_global_backend` themself?

Reply:
I thought about this kinda from the multiprocessing perspective: If imports are the only things that register backends, then we don't need to persist state since it'll be done at import, if my understanding of the situation is correct.

In any case, I can change this example and we can deal with it later.